src/observers/DocsLoggerObserver.ts

```diff
@@ -0,0 +1,121 @@
+import { OutputChannel } from "vscode";
+import { BaseEvent, PlatformInfoRetrieved, PackageInstallStart, PackageInstallSucceeded, PackageInstallFailed, DownloadStart, DownloadProgress, DownloadSizeObtained, DownloadValidating, DownloadIntegrityCheckFailed, ZipFileInstalling } from "../common/loggingEvents";
+import { EventType } from "../common/EventType";
+
+export class DocsLoggerObserver {
+    private downloadProgressDot: number;
+    constructor(private logger: OutputChannel) { }
+
+    public eventHandler = (event: BaseEvent) => {
+        switch (event.type) {
+            // Runtime Dependency
+            case EventType.DependencyInstallStart:

```

Comment:
What is the benefit of using this approach? It appears to add a lot of duplicated code and make thing more complex.

Reply:
1. Decouple different component
2. Make input/output of component more clear so it is easier to test

---
In reply to: [362360099](https://github.com/docascode/vscode-docs-build/pull/9#discussion_r362360099) [](ancestors = 362360099)